### ISSUE TITLE
macOS Sierra cannot find any cast devices

### ISSUE THREAD

**Issue by Olshansk** [muammar/mkchromecast#54]
After installing mkchromecast via brew, the application continuously searches and cannot find any cast devices. However, it's worth noting that this did work the very first time I used it.

After installing the application from source and running `python mkchromecast.py`, the following error shows up: `No handlers could be found for logger "pychromecast.socket_client"`.

Any insight on what the issue might be or what I should do to further debug?

**Comment by muammar** [comment]
Which version of pychromecast have you installed?. There are other people having similar issues (see https://github.com/balloob/pychromecast/issues?utf8=%E2%9C%93&q=No%20handlers%20could%20be%20found%20for%20logger%20) but I am not able to find any conclusive message. In some of the comments, it is mentioned that the `cast` object has to wait until the connection is fully setup. That is assured in [mkchromecast/cast.py#L243](https://github.com/muammar/mkchromecast/blob/master/mkchromecast/cast.py#L243). 

Does this happen both in LAN and WiFi?.

**Comment by Olshansk** [comment]
I'm using version 0.7.7 of PyChromecast as specified in the requirements file.

I tried updating to the latest release (0.8.1) and am getting the following error instead `ERROR:pychromecast.socket_client:Failed to connect, retrying in 5.0`

I've only tried this with Wifi.

**Comment by muammar** [comment]
> I tried updating to the latest release (0.8.1) and am getting the following error instead ERROR:pychromecast.socket_client:Failed to connect, retrying in 5.0

One last question, is it only happening when using mkchromecast?. I used to have these problems before, and my wifi router was to blame. I might be wrong. 

>I've only tried this with Wifi.

Does it happen with the cable?

**Comment by muammar** [comment]
Were you able to solve this problem?. I still think is something related to your wifi router. Some time ago, there someone reported in #27 that was not able to cast anything through wifi and updating his wifi driver helped https://github.com/muammar/mkchromecast/issues/27#issuecomment-291457757. Of course,  I understand that in macOS that is not the case, but I haven't been able to reproduce this issue. 

I had it before, and it was due to the router that was given to me by ISP.

**Comment by Olshansk** [comment]
It is indeed only happening with mkchromecast and it doesn't work when I connect to the router via ethernet. 

I'm planning to investigate more extensively but haven't gotten around to it yet.

**Comment by muammar** [comment]
That would be nice. Unfortunately, I haven't been able to reproduce it by myself.

**Comment by Olshansk** [comment]
tl;dr I got it to work by disabling AirServer but I also think we should set the `tries` argument in the `pychromecast.get_chromecasts` call.

Alright, I got it to work and have some interesting findings to share.

The reason it wasn't working is because I am running [AirServer](https://www.airserver.com/), which configures my laptop to be an AirPlay/Google Cast receiver. Disabling the Google Cast feature fixes the problem.

I haven't looked into the Google Cast protocol extensively, but it seems that AirServer's implementation is incomplete. Nmapping my google home and my chromecast shows that the two devices expose an http server on port 8008 and an ajp13 server on 8009. However, when I startup AirServer, while there is an ajp13 server on port 8009, nothing is exposed on 8008. This port is necessary to query the device status (see [here](https://github.com/balloob/pychromecast/blob/e39fa31029459784b2e78b5a68efddedc7bb5c42/pychromecast/dial.py#L13)), and returns an error otherwise. I also confirmed that a webserver on port is 8008 is indeed necessary according to [this documentation](https://github.com/jloutsenhizer/CR-Cast/wiki/Chromecast-Implementation-Documentation-WIP).

The reason mkchromecast wasn't working is because the call to `pychromecast.get_chromecasts` [here](https://github.com/muammar/mkchromecast/blob/9aa1ab97e38f2951527982b3dd985b872b2ebc04/mkchromecast/cast.py#L101) doesn't specify any configurations. This means that while pychromecast was able to successfully connect to two of my cast devices (home and chromecast), it failed to connect to my third cast device (airserver on laptop) and therefore never returns. 

The fix is very simple: set the `tries` argument to something small when making the `get_chromecasts` call. This would prevent the underlying library from retrying to connect to server in perpetuity and return the fully functioning and available cast devices. A value of 1 or 2 is sufficient in my opinion.

If you think it's a good idea, I will hapilly make the PR.

**Comment by muammar** [comment]
Thanks for this information!. I also think it would be a good idea to set `tries` to some finite value. If you do the PR, could you add it as an option? in that way, users can turn it to original behavior when desired. All flags in `mkchromecast` are set here https://github.com/muammar/mkchromecast/blob/master/mkchromecast/__init__.py (no idea if it was the best way of doing so), I don't know what name we could choose for that flag maybe just `--tries` or something else?.

**Comment by Olshansk** [comment]
Yup, that sounds good to me. I should have something up next weekend.

**Comment by muammar** [comment]
> Yup, that sounds good to me. I should have something up next weekend.
 
Thanks :).
